game/fs.cc

```diff
@@ -46,139 +44,55 @@ namespace {
-			#ifdef _WIN32
-				ITEMIDLIST* pidl;
-				HRESULT hRes = SHGetSpecialFolderLocation(nullptr, CSIDL_APPDATA|CSIDL_FLAG_CREATE, &pidl);
-				if (hRes != NOERROR) throw std::runtime_error("Unable to determine where Application Data is stored");
-				char p[MAX_PATH];
-				SHGetPathFromIDList(pidl, p);
-				conf = p;
-			#else
-				char const* p = getenv("XDG_CONFIG_HOME");
-				conf = (p ? p : home / ".config");
-			#endif
-				conf /= performous;
-			}
-			// Data
-			{
-			#ifdef _WIN32
-				data = conf;
-			#else
-				char const* p = getenv("XDG_DATA_HOME");
-				data = (p ? p / performous : home / ".local" / SHARED_DATA_DIR);
-			#endif
-			}
-			// Cache
-			{
-			#ifdef _WIN32
-				cache = data / "cache";  // FIXME: Should we use GetTempPath?
-			#else
-				char const* p = getenv("XDG_CACHE_HOME");
-				cache = (p ? p / performous : home / ".cache" / performous);
-			#endif
-			}
-			pathInit();
-		}
-		/// Initialize/reset data dirs (search path).
-		void pathInit() {
-			bool bootstrapping = paths.empty();  // The first run (during bootstrap)
-			if (!bootstrapping) {
-				std::string logmsg = "fs/info: Found system paths:\n";
-				logmsg += "  base:     " + base.string() + '\n';
-				logmsg += "  share:    " + share.string() + '\n';
-				logmsg += "  locale:   " + locale.string() + '\n';
-				logmsg += "  sysConf:  " + sysConf.string() + '\n';
-				logmsg += "  home:     " + home.string() + '\n';
-				logmsg += "  config:   " + conf.string() + '\n';
-				logmsg += "  data:     " + data.string() + '\n';
-				logmsg += "  cache:    " + cache.string() + '\n';
-				std::clog << logmsg << std::flush;
-			}
-			// Data dirs
-			std::string logmsg = "fs/info: Determining data dirs (search path):\n";
-			{
-				Paths dirs;
-				dirs.push_back(data);  // Adding user's data dir
-				dirs.push_back(share);  // Adding system data dir (relative to performous executable or PERFORMOUS_ROOT)
-			#ifndef _WIN32
-				// Adding XDG_DATA_DIRS
-				{
-					char const* xdg_data_dirs = getenv("XDG_DATA_DIRS");
-					std::istringstream iss(xdg_data_dirs ? xdg_data_dirs : "/usr/local/share/:/usr/share/");
-					for (std::string p; std::getline(iss, p, ':'); dirs.push_back(p / performous)) {}
-				}
-			#endif
-				// Adding paths from config file (during bootstrap config options are not yet available)
-				if (!bootstrapping) {
-					auto const& conf = config["paths/system"].sl();
-					for (std::string const& dir: conf) dirs.splice(dirs.end(), pathExpand(dir));
-				}
-				// Check if they actually exist and print debug
-				paths.clear();
-				std::set<fs::path> used;
-				for (auto dir: dirs) {
-					dir = fs::absolute(dir);
-					if (used.find(dir) != used.end()) continue;
-					logmsg += "  " + dir.string() + '\n';
-					paths.push_back(dir);
-					used.insert(dir);
-				}
-			}
-			if (!bootstrapping) std::clog << logmsg << std::flush;
-		}
+		
+	#if (BOOST_OS_WINDOWS)
+	#include "platform/fs_paths.win.inc"
+	#else
+	#include "platform/fs_paths.unix.inc"
```

Comment:
in game/exename.cc you include a lot of platforms, but here you only specify windows and unix? this looks inconsistent to me.

Reply:
It's because execname.cc actually had that many different platform implementations already in the code.

For my version, I did only windows and unix because that's the only two meaningful differences. The further differences between linux and osx, for example, are actually determined at runtime instead of compile-time.